### PR TITLE
add: eventgrid_domain data source, test, docs

### LINKED ISSUE #13008: Support for Data source: azurerm_eventgrid_domain
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When creating an application that can publish events to an Event Grid Domain, you might not own the deployment of the Event Grid Domain and can therefore not read any of the exported attributes such as `primary_access_key` that is needed in an app to authenticate to the Event Grid Domain. A solution would be to add `azurerm_eventgrid_domain` as a data source that exposes these values as exported attributes. The only solution I have found today is to run a snippet of ARM template to get the keys. This would eliminate the need for the ARM template.

### New or Affected Resource(s)

* azurerm_eventgrid_domain

* #0000

### PR DESCRIPTION
**Pull request by Hjortsberg**
This is a proposal to fix #13008.

The implementation is untested.
This is my first PR to this repo, please be constructive in the feedback so that I can improve. :)

### CODE REVIEW

File: internal/services/eventgrid/eventgrid_domain_data_source.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: EventGrid Domain %s (Resource Group %s) was not found: %+v", name, resourceGroup, err)
+		}
+
+		return fmt.Errorf("making Read request on EventGrid Domain '%s': %+v", name, err)
+	}
+
+	d.SetId(*resp.ID)
```
Comment (katbyte) [inline internal/services/eventgrid/eventgrid_domain_data_source.go:129]: we should be nil checking ID here

File: internal/services/eventgrid/eventgrid_domain_data_source_test.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+)
+
+type EventGridDomainDataSource struct {
+}
+
+func TestAccEventGridDomainDataSource_basic(t *testing.T) {
```
Comment (katbyte) [inline internal/services/eventgrid/eventgrid_domain_data_source_test.go:14]: we can probably remove the other tests, usually we just have a single data source test

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the pr @Hjortsberg - overall looks good and i've left some comments inline and the linters have called out the schema changes required

**Comment by stephybun** [comment]
@Hjortsberg since we haven't heard anything back yet i'm going to jump in here and finish up this PR so that your work can be merged and released to the community.

**Comment by Hjortsberg** [comment]
@stephybun, sorry for the lack of replies from me. I greatly appreciate the contribution here, thank you.

**Review by katbyte — APPROVED** [review]
Thanks @Hjortsberg & @stephybun - LGTM 🍰

**Pull request merged**